### PR TITLE
Petsy

### PR DESCRIPTION
**Pull request by jweeber**
https://jd-petsy.herokuapp.com/

Project with Jess, Deirdre, and Alysia.

Gems in Gemfile.

Baked Potato Costume is amazing.

### CODE REVIEW

File: app/assets/javascripts/home.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (hougland) [inline app/assets/javascripts/home.coffee:3]: Best practice is to delete files like this that you never use.

File: config/routes.rb [hunk 2]
```diff
@@ -1,56 +1,38 @@
...
+
+  get 'orders/order_fulfillment' => 'orders#index', as: 'order_fulfillment'
+  patch '/orderitems/:id/shipped' => 'orderitems#shipped', as: 'shipped'
+
+  resources :sessions, :only => [:new, :create]
+  get "/logout" => "sessions#destroy"
+  get "/orders/:id/checkout" => 'orders#checkout', as: 'order_checkout'
+  patch "/orders/:id/checkout" => "orders#checkout"
+  get '/orders/:id/confirmation' => 'orders#confirmation', as: "order_confirmation"
+
```
Comment (hougland) [inline config/routes.rb:37]: It may be helpful to add comments above each chunk of routes, so it's easier to see at a glance which routes are used for what.

File: app/controllers/orders_controller.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    params.permit(orderitem: [:quantity])
+  end
+
+  def order_update_params
+    params[:order][:credit_card_number] = params[:order][:credit_card_number][-4..-1]
+    params.permit(order: [:name_on_credit_card, :user_id, :city, :state, :billing_zip,
+      :email, :status, :street_address, :credit_card_cvv, :credit_card_number, :credit_card_exp_date])
+  end
+
+end
```
Comment (hougland) [inline app/controllers/orders_controller.rb:87]: I can't add things to my cart. Not sure exactly why, but it looks like it's generating a new order each time I click "add to cart" and each time I click "checkout." For example, I add "ballerina guinea" to my order, which takes me to "/orders/684/edit". When I click "proceed to checkout," it takes me to "/orders/686/edit", which is an empty cart. When I try to add another item, it takes me to "/orders/702/edit".

File: app/controllers/orders_controller.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:show_seller_orders]
+
+  def show
+    @orders = Order.find(current_order.id).orderitems
+  end
```
Comment (hougland) [inline app/controllers/orders_controller.rb:6]: Where is this called? I see that you have a route for it, but I can't find a view. I also get an error when I try to render "/orders/703/". I'm trying to figure out why you'd want to set @orders to be equal to an order's orderitems.

File: app/views/orders/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="order-info">
+      <strong><%= link_to Product.find(orderitem.product_id).name, product_path(orderitem.product_id) %></strong>
+      <br>Total Price $<%=number_with_precision orderitem.total_price, precision: 2 %>
+      <% @cart_price += orderitem.total_price %>
+      <%= form_for orderitem do |f| %>
+      <% @quantity_numbers = (1..orderitem.product.quantity).to_a %>
+        <%= f.label "Quantity" %>
+        <%= f.select 'quantity', options_for_select(@quantity_numbers, orderitem.quantity) %><br>
+        <%= f.submit "Update Quantity", class: "btn btn-info" %>
+      <% end %>
```
Comment (hougland) [inline app/views/orders/edit.html.erb:18]: Instead of iterating over `@order.orderitems`, you can iterate over @orderitems, which is defined in the controller as `Order.find(current_order.id).orderitems`

File: app/controllers/orders_controller.rb [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+  def show_seller_orders
+  	@user = User.find(current_user.id)
+    @user_orders_hash = Orderitem.where(user: current_user).group_by(&:order_id)
```
Comment (hougland) [inline app/controllers/orders_controller.rb:10]: I like how you build this hash!

File: app/controllers/orders_controller.rb [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    @revenue = @user.revenue
+    @completed_revenue = @user.revenue_by_status("Completed")
+    @pending_revenue = @user.revenue_by_status("Pending")
+    @completed_count = @user.order_by_status("Completed")
+    @pending_count = @user.order_by_status("Pending")
```
Comment (hougland) [inline app/controllers/orders_controller.rb:15]: Neat how you've built this method in a model, and are calling it here.

File: app/controllers/orders_controller.rb [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  end
+
+
+  def edit
+    @order  = Order.find(params[:id])
+    @orderitems = Order.find(current_order.id).orderitems
+  end
+
+  def update
+    @order = current_order
```
Comment (hougland) [inline app/controllers/orders_controller.rb:30]: I'm noticing that in some places you do `@order = current_order` and in some you do `@order = Order.find(params[:order_id])`. Is there a reason you're switching between them?

File: app/controllers/orders_controller.rb [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+      quantity_being_bought = item.quantity
+      available_quantity = product.quantity
+      new_quantity = available_quantity - quantity_being_bought
+      if new_quantity >= 0
+        product.update(quantity: new_quantity)
+      else
+        return false
+      end
+    end
+  end
```
Comment (hougland) [inline app/controllers/orders_controller.rb:75]: I like that this is a private method. Though, could this logic go into the order model, rather than the controller?

Comment (hougland) [inline app/controllers/orders_controller.rb:75]: Renaming this method may also be helpful. Since it returns a boolean, you may name it something like `items_successfully_removed?`

File: app/controllers/orders_controller.rb [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+    continue = remove_items_from_stock(@orderitems)
+    unless continue == false
+      @order.update(order_update_params[:order])
+    end
+    if @order.status == "Completed"
+      redirect_to order_confirmation_path(@order.id)
+    else
+      flash[:notice] = "Sorry! An item you wanted is out of stock. Check to see if you have duplicate items in your cart."
+      redirect_to edit_order_path(current_order.id)
+    end
```
Comment (hougland) [inline app/controllers/orders_controller.rb:41]: I think this logic could be cleaned up, and stuck into one if/else/elsif block, rather than an unless and if/else block. If anything, just to make it more clear to your readers what's happening :)

File: app/controllers/orders_controller.rb [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        product.update(quantity: new_quantity)
+      else
+        return false
+      end
+    end
+  end
+
+  def orderitem_edit_params
+    params.permit(orderitem: [:quantity])
+  end
```
Comment (hougland) [inline app/controllers/orders_controller.rb:79]: Was this accidentally copied over from the orderitems controller? I don't see it being called.

File: app/views/orders/checkout.html.erb [hunk 12]
```diff
@@ -0,0 +1,64 @@
+<h1>Complete your purchase:</h1>
+
+<h4>To finish up, please provide all of the following information:</h4><br>
+<%= form_for @order, html: { :class => "form-horizontal center" } do |f| %>
+
+  <div class="form-group">
+  <%= f.label :email, class: "col-xs-4 control-label required" %>
+  <div class="col-xs-4">
+  <%= f.email_field :email, class: "form-control", required: true %>
```
Comment (hougland) [inline app/views/orders/checkout.html.erb:9]: These custom form fields are so useful and great!

File: app/views/orders/confirmation.html.erb [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+Subtotal: $<%= number_with_precision item.total_price , precision: 2%><br><br>
+<% @cart_price += item.total_price %>
+<% end %>
+
+<h3> Total Price of Order: $<%= number_with_precision @cart_price , precision: 2   %></h3>
+<h3> Order Status: <%= @order.status %></h3>
+<h3> Time Order was placed: <%= @order.updated_at %></h3>
+
+<%= link_to "Thanks for Shopping at Petsy!", root_path %>
+</div>
```
Comment (hougland) [inline app/views/orders/confirmation.html.erb:23]: Not sure if this is github messing things up, but make sure you're indenting your html correctly.

File: app/views/orders/confirmation.html.erb [hunk 14]
```diff
@@ -0,0 +1,23 @@
+<h1> Congratulations! You bought some stuff!</h1>
+<div class="checkout">
+<h2>Details of Order # <%= @order.id %><br></h2>
+Sold To: <%= @order.name_on_credit_card %><br>
+Shipping To:
+<%= @order.street_address %><br>
+<%= @order.city %><br>
+<%= @order.state %>, <%= @order.billing_zip %><br>
+Credit Card : **** **** **** **** <%= @order.credit_card_number %><br><br>
```
Comment (hougland) [inline app/views/orders/confirmation.html.erb:9]: I like how you save/show only the last 4 digits and display them like this.

File: app/views/orders/confirmation.html.erb [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+<% @cart_price = 0 %>
+<% @order.orderitems.each do |item| %>
+Item: <%= link_to Product.find(item.product_id).name , product_path(Product.find(item.product_id).id) %><br>
+Quantity: <%= item.quantity %><br>
+Subtotal: $<%= number_with_precision item.total_price , precision: 2%><br><br>
+<% @cart_price += item.total_price %>
```
Comment (hougland) [inline app/views/orders/confirmation.html.erb:15]: Try not to do logic like this in the views. This should be a number passed to the view by a controller (logic could even happen in a model).

File: app/views/orders/confirmation.html.erb [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+<% end %>
+
+<h3> Total Price of Order: $<%= number_with_precision @cart_price , precision: 2   %></h3>
```
Comment (hougland) [inline app/views/orders/confirmation.html.erb:18]: yeahhhhhh number_with_precision is useful :)

File: app/views/orders/order_deets.html.erb [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+      <th>Item Name</th>
+      <th>Price</th>
+      <th>Quantity</th>
+      <th>Total Price</th>
+      <th>Shipped?</th>
+    </tr>
+  </thead>
+
+  <tbody>
+      <% orderitems = @user_orders_hash[(params[:order_id].to_i)] %>
```
Comment (hougland) [inline app/views/orders/order_deets.html.erb:15]: Is this because one user can have multiple orders?

File: app/views/orders/seller_items.html.erb [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+  <thead>
+    <tr>
+      <th>Item</th>
+      <th>Quantity</th>
+      <th>Item Price</th>
+      <th>Item Total</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% order_items = Orderitem.where(order_id: @order.id) && Orderitem.where(seller_id: user_id) %>
```
Comment (hougland) [inline app/views/orders/seller_items.html.erb:12]: Moar logic that should go in the controller :)

File: app/views/orders/seller_items.html.erb [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+    <% orders = order_items.collect(&:order_id).uniq %>
+    <% orders.each do |order| %>
+      <% orders = Orderitem.where(order_id: order.id) %>
+      <tr>
+      <% orders.each do |order| %>
+        <td><%= order.product.name %></td>
+        <td><%= order.quantity %></td>
+        <td>$<%= order.price %>0</td>
+        <td>$<%= order.price * order.quantity %>0</td>
```
Comment (hougland) [inline app/views/orders/seller_items.html.erb:21]: Why is there a 0 here? Maybe check out some rails methods that convert pennies into dollars - I think you use these in other places in the project :)

File: app/models/orderitem.rb [hunk 20]
```diff
@@ -0,0 +1,11 @@
+class Orderitem < ActiveRecord::Base
+  belongs_to :product
+  belongs_to :order
+  belongs_to :user
+  validates :quantity, presence: true, numericality: {only_integer: true, greater_than: 0}
+
+  def total_price
+    self.product.price * self.quantity
+  end
```
Comment (hougland) [inline app/models/orderitem.rb:9]: I like that you put a method in here :D

File: app/controllers/orderitems_controller.rb [hunk 21]
```diff
@@ -0,0 +1,45 @@
+class OrderitemsController < ApplicationController
+
+  def index
+    redirect_to order_path(current_order.id)
+  end
```
Comment (hougland) [inline app/controllers/orderitems_controller.rb:5]: Why do you need this method, since it only redirects?

File: app/controllers/orderitems_controller.rb [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+
+  def create
+    @product = Product.find(params[:product_id])
+    @orderitem = current_order.orderitems.create(product: @product, quantity: params[:quantity], user_id: @product.user_id)
+    @orderitem.update(price: @orderitem.total_price)
+    redirect_to edit_order_path(current_order.id)
+  end
```
Comment (hougland) [inline app/controllers/orderitems_controller.rb:12]: I don't think your variables need to be instance variables (with the @ in front of them), since they're only used locally within the controller.

File: app/controllers/orderitems_controller.rb [hunk 23]
```diff
@@ -0,0 +1,45 @@
+class OrderitemsController < ApplicationController
+
+  def index
+    redirect_to order_path(current_order.id)
+  end
+
+  def create
+    @product = Product.find(params[:product_id])
+    @orderitem = current_order.orderitems.create(product: @product, quantity: params[:quantity], user_id: @product.user_id)
+    @orderitem.update(price: @orderitem.total_price)
```
Comment (hougland) [inline app/controllers/orderitems_controller.rb:10]: I feel like lines 9 and 10 could be combined somehow?

File: app/controllers/orderitems_controller.rb [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  def update
+    @orderitem = Orderitem.find(params[:id])
+    if @orderitem.update(orderitem_edit_params[:orderitem])
+      @orderitem.update(price: @orderitem.total_price)
+      redirect_to edit_order_path(current_order.id)
+    else
+      redirect_to root_path
+    end
```
Comment (hougland) [inline app/controllers/orderitems_controller.rb:21]: I don't understand what this if/else block is doing.

File: app/controllers/home_controller.rb [hunk 25]
```diff
@@ -0,0 +1,9 @@
+class HomeController < ApplicationController
+	def index
+		@products = Product.where(retired: false)
+		@product_picker = Product.where(retired: false).where.not(id: nil).sample(3)
```
Comment (hougland) [inline app/controllers/home_controller.rb:4]: Nice! I like the sample(3) method.
Why do you need to specify that you're picking products where id != nil? Shouldn't every product have an id?

File: app/controllers/home_controller.rb [hunk 26]
```diff
@@ -0,0 +1,9 @@
...
+		@animals = Product.uniq.pluck(:animal)
+		@categories = Product.uniq.pluck(:category)
```
Comment (hougland) [inline app/controllers/home_controller.rb:6]: Ahh pluck is exciting too!

File: config/routes.rb (hunk 2 above)
Comment (hougland) [inline config/routes.rb:37]: I'm not totally sure, but it seems like there are routes you never use, especially within resources :orders and resources :orderitems.

**Comment by hougland** [comment]
WOO HOO!! YOU FINISHED BETSY!!!
Betsy was definitely one of the harder projects for me - figuring out all the routing, how the models interact, and how to split up work. Betsy kinda sucks 😛  But you did it!! 🎉 🎉🎉

I hope some of my bazillion comments are actually useful for you. Feel free to comment back if you have any questions. Sorry if my comments are overwhelming. I intend them to be helpful and productive, not fussy and annoying. My biggest takeaways would be to 1) add comments to describe parts of your code that may not be obvious to a reader (like the routes and controller actions that aren't standard) and 2) separate out more logic from controllers and views and stick it in models.

Your website is super pretty, and flows really well. I also luff the pet theme, images, and styling! 🐈 🐈 🐈  Awesome job!

File: app/models/review.rb [hunk 27]
```diff
@@ -0,0 +1,7 @@
+class Review < ActiveRecord::Base
+	belongs_to :product
+	belongs_to :user
+	validates :rating, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 5 }
+
```
Comment (kdefliese) [inline app/models/review.rb:5]: Good use of model validations!

File: app/views/products/show.html.erb [hunk 28]
```diff
@@ -0,0 +1,47 @@
...
+        <p>Reviewed By: <%= User.find(review.user_id).full_name %></p>
+        <% end %>
+        <p>Review Date: <%= review.created_at.strftime("%m/%d/%Y at %I:%M%p") %></p>
+        <p>Rating: <%= review.rating %>/5</p>
+        <span class="review-description"><p><%= review.description %></p></span>
+        <br>
+        <hr>
+        <% end %>
+        <% end %>
+    </div>
```
Comment (kdefliese) [inline app/views/products/show.html.erb:21]: Code inside your if/else/etc. blocks in your views should be indented!

File: app/controllers/products_controller.rb [hunk 29]
```diff
@@ -0,0 +1,91 @@
...
+		@products = Product.where(category: params[:category]).where(retired: false)
+	end
+
+	def new
+		@user = User.find(params[:id])
+		@product = @user.products.new
+		@category = Product.uniq.pluck(:category)
+		@animal = Product.uniq.pluck(:animal)
+		@category << "Create a Category"
+	end
```
Comment (kdefliese) [inline app/controllers/products_controller.rb:38]: This is a great way to make sure your dropdowns always stay up to date! Love it.

File: app/controllers/products_controller.rb [hunk 30]
```diff
@@ -0,0 +1,91 @@
...
+
+	def product_create_params
+
+		params.permit(product: [:name, :description, :price, :quantity, :category, :new_category, :photo_url, :user_id, :animal])
+	end
+
+	def product_update_params
+		params.permit(product: [:name, :description, :price, :quantity, :animal, :category, :photo_url, :user_id])
+	end
+
```
Comment (kdefliese) [inline app/controllers/products_controller.rb:90]: You don't need two sets of permitted params unless you have some particular reason for blocking specific attributes from being accessible upon update.

File: app/controllers/products_controller.rb [hunk 31]
```diff
@@ -0,0 +1,91 @@
...
+		end
+		if @product.save
+    	redirect_to user_product_path(current_user.id)
+  	else
+  		render :new
+  	end
+	end
+
+	def edit
+    @product = Product.find(params[:id])
```
Comment (kdefliese) [inline app/controllers/products_controller.rb:53]: Watch your indentation!

File: app/views/products/index.html.erb [hunk 32]
```diff
@@ -0,0 +1,25 @@
...
+  	<p class=retired-empty>OUT OF STOCK!</p>
+  <% else %>
+    <p class="order-quantity"><%= product.quantity %> Available!</p>
+
+	<%= submit_tag "Add to Cart", class: 'btn btn-primary' %><br>
+	<% end %>
+<hr>
+		<% end %>
+</div>
+	<% end %>
```
Comment (kdefliese) [inline app/views/products/index.html.erb:25]: Woah, this code is jumping all over the page! Sorry to keep harping on this but proper indentation makes unfamiliar code so much easier to read, it's best to get in the habit as soon as you can.

File: app/views/products/show_seller_products.html.erb [hunk 33]
```diff
@@ -0,0 +1,43 @@
+<h1>Current Products</h1>
+<%= link_to 'Add a New Product', new_product_path,  class: "store-links" %>
+<%= link_to 'Your Current Products', user_product_path, class: "store-links" %>
+<br>
+<% if @products.empty? %>
+	<h2>You have no products! Add a product to get your store started.</h2>
+<% else %>
```
Comment (kdefliese) [inline app/views/products/show_seller_products.html.erb:7]: Good job handling edge cases like this one throughout the project.

File: test/models/review_test.rb [hunk 34]
```diff
@@ -0,0 +1,31 @@
...
+  test "rating is valid if it is an integer" do
+    review = Review.new(rating: 3)
+    assert review.valid?
+  end
+
+  test "rating must be between 1 and 5" do
+    review = Review.new(rating: 7)
+    assert_not review.valid?
+    assert review.errors.keys.include?(:rating), "rating is not in the errors hash"
+  end
```
Comment (kdefliese) [inline test/models/review_test.rb:27]: Good job with your tests!

File: app/controllers/sessions_controller.rb [hunk 35]
```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.log_in(params[:username], params[:password])
```
Comment (annaleeherrera) [inline app/controllers/sessions_controller.rb:6]: Not sure if params password is necessary here bc it's encrypted- might want to double check though.

File: app/controllers/sessions_controller.rb [hunk 36]
```diff
@@ -0,0 +1,24 @@
...
+    order = Order.create
+    order.update(status: "Pending")
```
Comment (annaleeherrera) [inline app/controllers/sessions_controller.rb]: This  may be good to put into the orders controller.

File: app/controllers/users_controller.rb [hunk 37]
```diff
@@ -0,0 +1,22 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_create_params[:user])
+    if @user.save
+      redirect_to root_path
```
Comment (annaleeherrera) [inline app/controllers/users_controller.rb:10]: Good use or redirect and render!

File: app/views/sessions/new.html.erb [hunk 38]
```diff
@@ -0,0 +1,25 @@
+
+<h1>Sign In</h1>
+
+<div class="form-horizontal">
+<%= form_tag sessions_path, method: :post do %>
+
+<div class="form-group">
+ <%= label_tag :username, "Username:", class: "col-xs-4 control-label required" %>
+ <div class="col-xs-4">
+ <%= text_field_tag :username, "", class: "form-control" %>
```
Comment (annaleeherrera) [inline app/views/sessions/new.html.erb:10]: Indentation under classes will help readability.

File: app/views/users/new.html.erb [hunk 39]
```diff
@@ -0,0 +1,45 @@
...
+      <% @user.errors.each do |attribute, message| %>
+        <li><strong><%= attribute %></strong>: <%= message %></li>
+      <% end %>
+    </ul>
+  </div>
+<% end %>
+​
+<%= form_for @user, :html => { :class => "form-horizontal center" } do |f| %>
+  <div class="form-group">
+  <%= f.label :username, class: "col-xs-4 control-label required" %>
```
Comment (annaleeherrera) [inline app/views/users/new.html.erb:15]: Indent after divs

File: test/models/user_test.rb [hunk 40]
```diff
@@ -0,0 +1,64 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
+
+  test "validations: username can't be blank" do
+    user = User.new
+
+    assert_not user.valid?
+    assert user.errors.keys.include?(:username), "username is not in the errors hash"
```
Comment (annaleeherrera) [inline test/models/user_test.rb:9]: does this show up as an error on the User Interface?  If so, it could be written to be readable to a general user.

**Comment by annaleeherrera** [comment]
YOU SURVIVED BETSY!! I'm glad you are alive and well and didn't kill each other ;)  
Betsy was our first big group project and we learned how to work with different working styles.  If you were to take anything away from the project- take that!

I like that your code was simple and easy to read.  There was not much excess.  In the views, it could  be cleaned up a bit with indentation, I know it gets a bit confusing when you are implementing ruby code.  Try to keep any logic in the controllers vs. views. 

Good job committing frequently.  But also beware of over-committing bc it could get messy when you want to track your process.  You want to be able to clearly see what your process was, and have concise but descriptive commit messages.  

GREAT JOB TEAM!! KILLED IT.

File: app/controllers/orders_controller.rb (hunk 3 above)
Comment (jweeber) [inline app/controllers/orders_controller.rb:87]: Thanks, Ricky. We thought there might be something weird there. Sounds like the cart was doing some caching or something. Deirdre and I had tested on both our computers once we deployed and it was working fine, but others were saying it wasn't working so not entirely sure what happened. The groups currently using it for the shipping service project have fixed it! 🌟

File: app/controllers/orders_controller.rb (hunk 4 above)
Comment (jweeber) [inline app/controllers/orders_controller.rb:6]: Whoops, probably leftover from something we started. Should have been deleted. Thanks!

File: app/views/orders/edit.html.erb (hunk 5 above)
Comment (jweeber) [inline app/views/orders/edit.html.erb:18]: 👍🏼

File: app/controllers/orders_controller.rb (hunk 8 above)
Comment (jweeber) [inline app/controllers/orders_controller.rb:30]: Pressed for time and confusion? 😅

File: app/controllers/orders_controller.rb (hunk 9 above)
Comment (jweeber) [inline app/controllers/orders_controller.rb:75]: We wondered about that and now that we know more about where things should go you're right - this would be great to move to the model!

File: app/views/orders/order_deets.html.erb (hunk 17 above)
Comment (jweeber) [inline app/views/orders/order_deets.html.erb:15]: Yep! A merchant user can have multiple orders.

File: app/controllers/home_controller.rb (hunk 25 above)
Comment (jweeber) [inline app/controllers/home_controller.rb:4]: Yeaahh.. hmm I think I meant to put if the image was not nil, in case a user didn't add an image to their product.

File: app/controllers/products_controller.rb (hunk 31 above)
Comment (jweeber) [inline app/controllers/products_controller.rb:53]: Thanks! I get a little sloppy with it when pressed for time. Gotta watch that. 🙃

File: app/controllers/sessions_controller.rb (hunk 36 above)
Comment (jweeber) [inline app/controllers/sessions_controller.rb]: 👍🏼

**Comment by jweeber** [comment]
Thanks to all of you for your helpful comments! Going to keep these in mind going forward. We got put on the three-person team so we're just really proud to have completed the requirements given that it's a four-person project. We also had a lot fun finding pet costume pics. 🐈 😁
